Shop_Unity_base/Assets/Scripts/Shop/Model/Components/RPGStatsComponent.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Utility;
+
+namespace Model
+{
+    public partial class RPGStatsComponent : Component<CustomerModel>, ITargetable
+    {
+        private int health;
+        private int mana;
+        private int nourishment;
+        private int strength;
+
+        // Geters can be expanded to add logic before returning
+        public int Health => health;
+        public int Mana => mana;
+        public int Nourishment => nourishment;
+        public int Strength => strength;
+
+        private RPGStatsComponent(int health, int mana, int nourishment, int strength)
+        {
+            this.health = health;
+            this.mana = mana;
+            this.nourishment = nourishment;
+            this.strength = strength;
+        }
+
+        private RPGStatsComponent(RPGStatsComponent statsComponent)
+        {
+            CloneMembers(statsComponent);
+        }
+
+        private void CloneMembers(RPGStatsComponent statsComponent)
+        {
+            health = statsComponent.health;
+            mana = statsComponent.mana;
+            nourishment = statsComponent.nourishment;
+            strength = statsComponent.strength;
+        }
+
+        public override IClonable Clone()
+        {
+            return new RPGStatsComponent(this);
+        }
+
+        public void OnHit(StatsEffect effect, int effectValue, float duration = 0)
+        {
+            // Negate the value so it damages by default
+            OnEffect(effect, -effectValue, duration);
+        }
+
+        public void Consume(ConsumableComponent consumable)
+        {
+            OnEffect(consumable.Effect, consumable.Potency, consumable.Duration);
+        }
+
+        private void OnEffect(StatsEffect effect, int effectValue, float duration = 0)
+        {
+            switch (effect)
+            {
+                case StatsEffect.HealthModifier:
+                    AddHealth(effectValue);
+                    break;
+                case StatsEffect.ManaModifier:
+                    AddMana(effectValue);
+                    break;
+                case StatsEffect.NourishmentModifier:
+                    AddNourishment(effectValue);
+                    break;
+                case StatsEffect.StrengthModifier:
+                    AddStrength(effectValue, duration);
+                    break;
+                case StatsEffect.Invisibility:
+                    SetInvisible(duration);
+                    break;
+                default:
+                    throw new System.NotSupportedException("The effect of the consumable cannot be determined");
+            }
+        }
+
+        // Extra parameters may be added to these functions to handle effects 
+        // that apply over time or any other additional logic
```

Comment:
While working on component I realized that if I made all the code make sense for a proper implementation of game features, it would become pretty large.

So I opted for pseudocoding and indicating how it could be used. Not ideal, but I chose to go for simplicity and make it easier on myself.